ladybug/comfort/pmv.py

```diff
@@ -1,1227 +1,555 @@
-            x_missing = secant(low_bound, up_bound, fn, error)
-            if x_missing == 'NaN':
-                x_missing = bisect(low_bound, up_bound, fn, error)
-
-            missing_val.append(x_missing)
-
-        # Set the conditions of the comfort model to have the new missing_val.
-        if missing_input == 0:
-            self.__air_temperature = missing_val
-        elif missing_input == 1:
-            self.__rad_temperature = missing_val
-        elif missing_input == 2:
-            self.__wind_speed = missing_val
-        elif missing_input == 3:
-            self.__rel_humidity = missing_val
-        elif missing_input == 4:
-            self.__met_rate = missing_val
-        elif missing_input == 5:
-            self.__clo_value = missing_val
-        elif missing_input == 6:
-            self.__external_work = missing_val
-
-        # Tell the comfort model to recompute now that the new values have been set.
-        self.__isRecalcNeeded = True
-
-        if self.__singleVals is True:
-            return missing_val[0]
-        else:
-            return missing_val
+    Args:
+        ppd: The percentage of people dissatisfied (PPD) for which you want to know
+            the possible PMV.
+        ppd_error: The acceptable error in meeting the target PPD.  Default = 0.001.
+
+    Returns:
+        pmv_lower: The lower (cold) PMV value that will produce the input ppd.
+        pmv_upper: The upper (hot) PMV value that will produce the input ppd.
+    """
+    if ppd > 5 and ppd < 100:
+        pmv_low = -3
+        pmv_mid = 0
+        pmv_hi = 3
+
+        def fn(pmv):
+            return (
+                (100.0 - 95.0 *
+                 math.exp(-0.03353 * pow(pmv, 4.0) - 0.2179 * pow(pmv, 2.0))) - ppd)
+
+        # Solve for the missing lower PMV value.
+        pmv_lower = secant(pmv_low, pmv_mid, fn, ppd_error)
+        if pmv_lower == 'NaN':
+            pmv_lower = bisect(pmv_low, pmv_mid, fn, ppd_error)
+        # Solve for the missing higher PMV value.
+        pmv_upper = secant(pmv_mid, pmv_hi, fn, ppd_error)
+        if pmv_upper == 'NaN':
+            pmv_upper = bisect(pmv_mid, pmv_hi, fn, ppd_error)
+
+        return pmv_lower, pmv_upper
+    else:
+        raise ValueError(
+            'PPD value {}% is outside acceptable limits of the PMV model.'.format(ppd))
+
+
+def calc_missing_pmv_input(target_pmv, other_inputs,
+                           missing_pmv_input='air temperature',
+                           low_bound=0, up_bound=100, error=0.001):
+    """Return the value of a missing_pmv_input given a target_pmv and the 6 other inputs.
+
+    This is particularly useful when trying to draw comfort polygons on charts
+    using the PMV model.
+
+    Args:
+        target_pmv: The target pmv that you are trying to produce from the inputs to
+            this pmv object.
+        other_inputs: A list of values for the other 6 inputs of the PMV model that
```

Comment:
I think there are better ways to handle this. You can be explicit about what you need and accept a dictionary. I would also suggest to use `tolerance` instead of `error`.

``` def calc_missing_pmv_input(target_pmv, input_values, ... , tolerance=0.001):```

Input values will be a dictionary with all the values. The missing key is your missing input.

Reply:
Ok. So we just expect a dictionary of pmv inputs that is missing one of the values?  We might have to do a lot more checks of the keys as a result but I can do that.

Reply:
Ok. I have changed this over to a dictionary input with the result returning a dictionary that contains the missing key.